solutions/usaco-712.mdx

```diff
@@ -108,3 +108,45 @@ int main() {
 </CPPSection>
 
 </LanguageSection>
+
+## Alternative Solution
+
+For every cow (say cow $N$), we can check whether which cows have one entry/exit point between both of cow $N$'s entry and exit points. Then, we can add each of these cows as a pair with cows $N$ to a list of crossings, making sure we have no repeats. Then we can just output the length of this list of crossings to get our final answer.
+
+## Implementation
+	
+**Time Complexity:** $\mathcal{O}(T^3)$, where $T$ represents the total number of characters in Farmer John's string.
+
+<LanguageSection>
+    
+<PySection>
+    
+```python
+TOTAL = 52
+
+recordings = open('circlecross.in').read().strip()
+
+crossings = []
+
+for i in range(TOTAL):
+    if i == recordings.index(recordings[i]):
+        # First appearance of letter
+        continue
+        print(True)
+    else:
+        appearances = []
+        for j in range(recordings.index(recordings[i]) + 1, i):
+            if recordings[j] in appearances:
+                appearances.remove(recordings[j])
+            else:
+                appearances.append(recordings[j])
+        for k in appearances:
+            if [recordings[i], k] not in crossings and [k, recordings[i]] not in crossings:
+                crossings.append([recordings[i], k])
```

Comment:
this section could use some comments but idrc

Reply:
ok ill add some minor ones